File: apple/Sources/ReactNativeWatchOS/Modifiers/DefaultScrollAnchor.swift
import SwiftUI

/// SwiftUI `.defaultScrollAnchor(_:)`. Sets the unit point the scroll view
/// initially aligns to (and re-aligns to when content size changes).
/// `anchor` is parsed via the shared `parseUnitPoint` ('top' | 'center' |
/// 'bottom' | … | `{ x, y }`); defaults to `.center`.
///
/// Gated to watchOS 10+. Returns the view unchanged on older systems.
enum RNWDefaultScrollAnchorModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("defaultScrollAnchor") { view, params, _ in
            if #available(watchOS 10.0, *) {
                let anchor = parseUnitPoint(params["anchor"]) ?? .center
                return AnyView(view.defaultScrollAnchor(anchor))
            }
            return view
        }
    }
}

File: src/modifiers/defaultScrollAnchor.ts
import { createModifier } from './createModifier';
import type { UnitPoint } from '../components/LinearGradient';

export interface DefaultScrollAnchorParams {
  /// A named `UnitPoint` ('top' | 'center' | 'bottom' | …) or an explicit
  /// `{ x, y }` in unit space.
  anchor: UnitPoint;
}

/// SwiftUI `.defaultScrollAnchor(_:)`. Sets the initial anchor the scroll
/// view aligns to (and re-aligns to when content size changes). Gated to
/// watchOS 10+ natively; a no-op on older systems.
export function defaultScrollAnchor(
  anchor: UnitPoint
): ReturnType<typeof createModifier>;
export function defaultScrollAnchor(
  params: DefaultScrollAnchorParams
): ReturnType<typeof createModifier>;
export function defaultScrollAnchor(
  a: UnitPoint | DefaultScrollAnchorParams
) {
  if (typeof a === 'string' || ('x' in a && 'y' in a)) {
    return createModifier('defaultScrollAnchor', { anchor: a });
  }
  return createModifier('defaultScrollAnchor', a);
}
